pkg/roachprod/install/cockroach.go

```diff
@@ -405,9 +381,15 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
 	}
 
 	if c.allowServiceRegistration() {
-		err := c.maybeRegisterServices(ctx, l, startOpts, c.FindOpenPorts)
-		if err != nil {
-			return err
+		// Only register services when starting a virtual cluster, or using custom
+		// ports, or for local cluster port management to avoid collisions. The
+		// lookup logic will automatically fall back to the default ports if the
+		// service is not found (or has not been registered).
+		if customPortsSpecified() || c.IsLocal() || startOpts.Target != StartDefault {
```

Comment:
So IIUC, this is letting `SharedProcess` through, but then in the actual registration code, it becomes a no-op? I guess that's fine, I don't have a strong opinion on what level of the stack this decision is made, although it is slightly weird to me that the service type verification is split in this way.

Reply:
Yeah, `maybeRegisterServices` as the name implies already states it will only `maybe` register a service, I think it should be okay to remove the condition here, and add a comment on `maybeRegisterServices`?